common/src/main/java/com/wynntils/wc/utils/scoreboard/objectives/Objective.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.utils.scoreboard.objectives;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+
+/*
+ *  This class is an info holder for Wynncraft's daily and guild objectives
+ */
+public record Objective(String goal, int score, int maxScore, long updatedAt, String original) {
+    public static final Pattern OBJECTIVE_PARSER_PATTERN = Pattern.compile("^[- ] (.*): *(\\d+)/(\\d+)$");
+
+    public static Objective parseObjectiveLine(String objectiveLine) {
+        String stripped = ChatFormatting.stripFormatting(objectiveLine);
+
+        assert stripped != null;
+
+        Matcher matcher = OBJECTIVE_PARSER_PATTERN.matcher(stripped);
+        String goal = null;
+        int score = 0;
+        int maxScore = 0;
+
+        // Match objective strings like "- Slay Lv. 20+ Mobs: 8/140" or "- Craft Items: 0/6"
+        // Could also be multi line like:
+        // "- Trade 24E with"
+        // "  lvl 10 players: 0/24"
+        if (matcher.find()) {
+            goal = matcher.group(1);
+            try {
+                score = Integer.parseInt(matcher.group(2));
+                maxScore = Integer.parseInt(matcher.group(3));
+            } catch (NumberFormatException e) {
+                // Ignored, goal is already null
+            }
+        }
```

Comment:
I realize you are mostly copying code that I wrote myself but I don't understand how this can work. 😆 How can we ever parse a multi-line objective with this code?

Reply:
Yes, that was what I meant. From the comment:
```
        // Could also be multi line like:
        // "- Trade 24E with"
        // "  lvl 10 players: 0/24"
```

Reply:
But honestly, I think I wrote that comment, and I can't really see how my code in legacy could deal with this properly. :-& Or maybe it did originally, but it has been rewritten since so that this handling has broken.

Reply:
I think maybe Wynn changed it so they are single line only

Reply:
Maybe we should do something like what you are doing in the quest manager, and build up the entire scoreboard, as seen on the screen by the user, before starting to parse things out of it (like a List<String>, basically). And just listen to the events as signals that we need to update this in-memory parsed scoreboard, and if it changed, reparse the parts that changed. (party health, tracked quest, etc)

Reply:
I would rather confirm that multi line objectives still exist before making it unnecessary complicated, as for party tracking, we don't need to get the entire scoreboard

Reply:
I'm not sure how we can confirm that. Maybe @HeyZeer0 knows, or can find out. I only occasionally got those, the only one I can remember was the one urging you to go throw emeralds on noobs.

Reply:
Can you try explaining the benefits compared to now?

Reply:
the answer is that it never really did handle them properly - it only matched the second line, so the objective overlay would just display "lvl 10 players" as the objective. can't comment on whether the objective is still in the game or not

Reply:
It was changed to trade with newbies so it is 1 line. If it is not, I will fix it later.